pymeasure/experiment/results.py

```diff
@@ -129,16 +131,54 @@ def __init__(self, columns, delimiter=','):
         """
         super().__init__()
         self.columns = columns
+        self.units = self._parse_columns(columns)
         self.delimiter = delimiter
 
+    @staticmethod
+    def _parse_columns(columns):
+        """Parse the columns to get units in parenthesis."""
+        units_pattern = r"\((?P<units>[\w/\(\)\*\t]+)\)"
+        units = {}
+        for column in columns:
+            try:
+                match = re.search(units_pattern, column)
+            except TypeError:
```

Comment:
In what situation do you expect to get a TypeError?

Reply:
The regex part is taken from the parsing for the plotter (only with the regex pattern expanded).
I did not try to improve it.